src/dso_api/dynamic_api/serializers.py

```diff
@@ -54,9 +54,8 @@ def get_request(self):
         """
         return self.context["request"]
 
-    @property
-    def fields(self):
-        fields = super().fields
+    def get_fields(self):
```

Comment:
Controleer maar dat relatie filtering werkt ;)

Reply:
Verwacht het wel, dit wordt aangeroepen in de `Serializer` base class:

```
    @cached_property
    def fields(self):
        """
        A dictionary of {field_name: field_instance}.
        """
        # `fields` is evaluated lazily. We do this to ensure that we don't
        # have issues importing modules that use ModelSerializers as fields,
        # even if Django's app-loading stage has not yet run.
        fields = BindingDict(self)
        for key, value in self.get_fields().items():
            fields[key] = value
        return fields
```

ofwel, i.p.v. 'fields' overriden wordt nu 'get_fields()' overschreven.

Reply:
Hebben we daarnaast geen filtering tests op relatie filters?

Reply:
> Verwacht het wel, dit wordt aangeroepen in de `Serializer` base class:
> 
> ```
>     @cached_property
>     def fields(self):
>         """
>         A dictionary of {field_name: field_instance}.
>         """
>         # `fields` is evaluated lazily. We do this to ensure that we don't
>         # have issues importing modules that use ModelSerializers as fields,
>         # even if Django's app-loading stage has not yet run.
>         fields = BindingDict(self)
>         for key, value in self.get_fields().items():
>             fields[key] = value
>         return fields
> ```
> 
> ofwel, i.p.v. 'fields' overriden wordt nu 'get_fields()' overschreven.

Nu herken ik het goed, geen filtering maar authentificatie op velden niveuew wordt gedaan later (na dat request is binnen de scope van de serializer), en dus `cached_property` kan niet gebruikt worden (of moet cache weggegooid).

We hadden wel een test voor dit, geloof ik.

Reply:
> Authentificatie op velden niveuew wordt gedaan later (na dat request is binnen de scope van de serializer),

Ah, duidelijk! Per request wordt de serializer overnieuw gevuld. Alleen geld dat natuurlijk niet voor onderliggende velden van een sub-serializer, omdat die eenmalig gemaakt is.

Reply:
Niet alleen dat, maar ook dat onderliggende serializer is aangemaakt zonder request en later DRF doet request injectie van de request.